packages/form-builder/src/settings/context.js

```diff
@@ -0,0 +1,38 @@
+import React, {createContext, useContext} from 'react';
+
+export const FormSettingsContext = createContext();
+
+export const FormSettingsProvider = ({formSettings, setFormSettings, children}) => {
+
+    const setFormSettingsObserver = ( settings ) => {
+        console.log( settings )
+        setFormSettings( settings )
+    }
+
+    return (
+        <FormSettingsContext.Provider value={[formSettings, setFormSettingsObserver]}>
+            {children}
+        </FormSettingsContext.Provider>
+    )
+}
+
+export const useFormSettings = () => {
+    const [formSettings, setFormSettings] = useContext(FormSettingsContext)
+    const updateSetting = ( newSettings ) => {
+        setFormSettings({
+            ...formSettings,
+            ...newSettings,
+        })
+    }
+    return [
+        formSettings,
+        updateSetting,
+    ]
+}
+
```

Comment:
@kjohnson since you're now getting into updating your provider/context state, I would highly recommend replacing this `useState` implementation with `useReducer`.  It's much more tuned for updating store objects & has performance benefits because you can update specific nested properties.  It might seem like a little more to setup but will be very helpful in the long run and is much easier to manage.  

During GiveIO we relied on this setup heavily so you can use these references for a head start:

- [Context](https://github.com/impress-org/giveio/blob/develop/resources/js/app/team-dashboard/campaigns/stores/campaign.jsx)
- [Reducer](https://github.com/impress-org/giveio/blob/develop/resources/js/app/team-dashboard/campaigns/reducers/campaignReducer.js)
- [Implementation](https://github.com/impress-org/giveio/blob/develop/resources/js/app/team-dashboard/campaigns/sidebars/NameDescription.jsx)

Lmk if you have any questions

Reply:
My reasoning in reaching for the Context API and not a reducer is for simplicity. These settings are flat and we aren't working with nested/shaped data, so I didn't think a reducer would be beneficial.

Reply:
Gotchya, I do have to imagine form settings are going to get more complex down the road. 

FYI this would not replace the Context API.  The reducer would replace how you're using `useState`.  You're honestly already almost doing it with assigning the value of your provider with an initial state and way to be updated.  The `useReducer` route sites inside the provider and gives you a way to explicitly dispatch changes to your store which I found makes the logic more readable and easier to manage.

Reply:
Ok. That comes down to my lack of experience with reducers, I just remember them being difficult.

Reply:
Looking at these settings more, I think it would actually make sense to break them out into their own nested objects based on the group of settings to simplify data shape & naming conventions.

```
formSettings {
  donationGoal {
    enable,
    autoClose,
    amount,
    format
  }
}
```

Then with the reducer you could do something like:

```
dispatch(updateDonationGoalSettings({enable: true}))
```

Reply:
Hmm... I think that makes sense. We are coming from flat storage, but for NextGen the settings are stored as JSON, so that would be easy enough to change to shaped data.

Reply:
@jonwaldstein would you be okay with resolving this in a separate issue, which can be added to next cycle?

Reply:
@kjohnson yup sounds good to me